libs/vkd3d/device.c

```diff
@@ -4676,12 +4710,29 @@ static void d3d12_device_caps_init_feature_options5(struct d3d12_device *device)
 {
     const D3D12_FEATURE_DATA_D3D12_OPTIONS *options = &device->d3d12_caps.options;
     D3D12_FEATURE_DATA_D3D12_OPTIONS5 *options5 = &device->d3d12_caps.options5;
+    const struct vkd3d_physical_device_info *info = &device->device_info;
 
     options5->SRVOnlyTiledResourceTier3 = options->TiledResourcesTier >= D3D12_TILED_RESOURCES_TIER_3;
     /* Currently not supported */
     options5->RenderPassesTier = D3D12_RENDER_PASS_TIER_0;
-    /* Currently not supported */
-    options5->RaytracingTier = D3D12_RAYTRACING_TIER_NOT_SUPPORTED;
+
+    /* Currently disabled until fully supported, but add checks for now. */
+    if (info->ray_tracing_pipeline_features.rayTracingPipeline &&
+        info->acceleration_structure_features.accelerationStructure &&
+        info->ray_tracing_pipeline_properties.maxRayHitAttributeSize >= D3D12_RAYTRACING_MAX_ATTRIBUTE_SIZE_IN_BYTES &&
+        /* Group handle size must match exactly or ShaderRecord layout will not match.
+         * Can potentially fixup local root signature if HandleSize < 32,
+         * but Vulkan is essentially specced to match DXR directly. */
+        info->ray_tracing_pipeline_properties.shaderGroupHandleSize == D3D12_SHADER_IDENTIFIER_SIZE_IN_BYTES &&
+        info->ray_tracing_pipeline_properties.shaderGroupBaseAlignment <= D3D12_RAYTRACING_SHADER_TABLE_BYTE_ALIGNMENT &&
+        info->ray_tracing_pipeline_properties.shaderGroupHandleAlignment <= D3D12_RAYTRACING_SHADER_RECORD_BYTE_ALIGNMENT &&
+        info->ray_tracing_pipeline_properties.maxRayRecursionDepth >= D3D12_RAYTRACING_MAX_DECLARABLE_TRACE_RECURSION_DEPTH)
+    {
+        INFO("DXR could potentially be supported, but not enabling by default for now.\n");
+        options5->RaytracingTier = D3D12_RAYTRACING_TIER_NOT_SUPPORTED;
+    }
+    else
+        options5->RaytracingTier = D3D12_RAYTRACING_TIER_NOT_SUPPORTED;
```

Comment:
We should probably move this whole thing to some separate function like the tiled resources one, since there's going to be a fair bit of stuff to check for Tier 1.0 and 1.1.

Reply:
Fixed.